chassis/core/src/main/java/com/griddynamics/jagger/util/generators/WorkloadGenerator.java

```diff
@@ -42,6 +44,17 @@ static WorkloadClockConfiguration generateLoad(JLoadProfile jLoadProfile) {
             userGroupsClockConfiguration.setDelay(new FixedDelay(profileUserGroups.getDelayBetweenInvocationsInSeconds()));
             userGroupsClockConfiguration.setTickInterval(profileUserGroups.getTickInterval());
             clockConfiguration = userGroupsClockConfiguration;
+        } else if (jLoadProfile instanceof JLoadProfileInvocation) {
```

Comment:
you can apply patterns here. I mean, basically, in each if clause you generate an instance of WorkloadClockConfiguration. and each clause is for different JLoadProfile subtype. So, you can have a separate factory for each JLoadProfile subtype and in the method you just get that factory from map by JLoadProfile instance as a key and create an instance using a factory you got.